src/main/scala/io/glutenproject/integration/tpc/DataGen.scala

```diff
@@ -45,6 +44,8 @@ object DataGen {
     schema.fields.foreach { f =>
       if (typeMapping.containsKey(f.dataType)) {
         modifiers.add(typeMapping.get(f.dataType))
+      } else if (f.dataType.isInstanceOf[DecimalType]) {
```

Comment:
can this special case be avoid somehow

Reply:
No, because DecimalType id DecimalType(precision, scale), so typeMapping does not contain the TypeModifier